frontend/src/app/core/services/previous-route.service.ts

```diff
@@ -15,7 +15,7 @@ export class PreviousRouteService {
     this.previousUrl = null;
 
     this.router.events
-      .pipe(filter((event: RouterEvent) => event instanceof NavigationEnd))
+      .pipe(filter((event): event is NavigationEnd => event instanceof NavigationEnd))
```

Comment:
probably don't need an is guard here

```suggestion
      .pipe(filter((event) => event instanceof NavigationEnd))
```